File: ValorPalabra.pas
program ValorPalabra;
uses crt;
var
   palabra: string;
   i, total: integer;
begin
   clrscr;
   repeat
      total := 0;
      writeln('Ingrese una palabra:');
      readln(palabra);

      for i := 1 to length(palabra) do
      begin
         if (palabra[i] in ['A'..'Z']) then
            total := total + ord(palabra[i]) - ord('A') + 1
         else if (palabra[i] in ['a'..'z']) then
            total := total + ord(palabra[i]) - ord('a') + 1;
      end;

      writeln('Valor de la palabra: ', total);
   until total = 100;

   writeln('Programa finalizado (palabra con 100 puntos).');
   readln;
end.

File: ValorPalabra.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char palabra[100];
    int i, total;

    do {
        total = 0;
        printf("Ingrese una palabra:\n");
        scanf("%s", palabra);

        for(i = 0; i < strlen(palabra); i++) {
            char c = tolower(palabra[i]);
            if(c >= 'a' && c <= 'z')
                total += (c - 'a' + 1);
        }

        printf("Valor de la palabra: %d\n", total);
    } while(total != 100);

    printf("Programa finalizado (palabra con 100 puntos).\n");
    return 0;
}
